### PR TITLE
Support for BIP39 password argument

### PR DESCRIPTION
**Pull request by LeeAlephium**

### CODE REVIEW

File: lib/wallet.ts [hunk 1]
```diff
@@ -73,8 +73,8 @@ export const getPath = (addressIndex?: number) => {
   return `m/44'/${coinType}/0'/0/${addressIndex || '0'}`
 }
 
-export const getWalletFromMnemonic = (mnemonic: string): Wallet => {
-  const seed = bip39.mnemonicToSeedSync(mnemonic)
+export const getWalletFromMnemonic = (mnemonic: string, password = ''): Wallet => {
```
Comment (mvaivre) [inline lib/wallet.ts:76]: We're using `password` and `passphrase` interchangeably. Should we stick to one term, maybe `passphrase` so we don't confuse it with the wallet's password?

File: lib/wallet.ts [hunk 2]
```diff
@@ -141,28 +141,28 @@ export const deriveNewAddressData = (
...
-  const dataDecrypted = decrypt(password, encryptedWallet)
+export const walletOpen = (passphrase: string, encryptedWallet: string) => {
+  const dataDecrypted = decrypt(passphrase, encryptedWallet)
   const config = JSON.parse(dataDecrypted) as StoredState
 
   return getWalletFromMnemonic(config.mnemonic)
 }
 
-export const walletEncrypt = (password: string, mnemonic: string) => {
+export const walletEncrypt = (passphrase: string, mnemonic: string) => {
```
Comment (mvaivre) [inline lib/wallet.ts:163]: Isn't this actually the password? (the one we type to lock / unlock)?

Comment (LeeAlephium) [reply lib/wallet.ts:163]: x_x I thought I skipped over that one (I avoid mass find-replace exactly for this reason)

Comment (LeeAlephium) [reply lib/wallet.ts]: Ah it was the one in `encrypt`

File: lib/wallet.ts [hunk 3]
```diff
@@ -141,20 +141,20 @@ export const deriveNewAddressData = (
...
   if (!bip39.validateMnemonic(mnemonic)) {
     throw new Error('Invalid seed phrase')
   }
-  return getWalletFromMnemonic(mnemonic)
+  return getWalletFromMnemonic(mnemonic, passphrase)
 }
 
-export const walletOpen = (password: string, encryptedWallet: string) => {
-  const dataDecrypted = decrypt(password, encryptedWallet)
+export const walletOpen = (passphrase: string, encryptedWallet: string) => {
```
Comment (mvaivre) [inline lib/wallet.ts:156]: Isn't this the "password" as well?

**Review by mvaivre — APPROVED** [review]

**Pull request merged**